tests/publiccloud/xfsprepare.pm

```diff
@@ -73,15 +99,26 @@ sub create_config {
     ## Profile config file containing device and mount point definitions
     assert_script_run("echo 'export TEST_DIR=$mnt_xfs' >> $CONFIG_FILE");
     assert_script_run("echo 'export SCRATCH_MNT=$mnt_scratch' >> $CONFIG_FILE");
-    assert_script_run("echo 'export TEST_DEV=${device}1' >> $CONFIG_FILE");
-    assert_script_run("echo 'export SCRATCH_DEV=${device}2' >> $CONFIG_FILE");
+    assert_script_run("echo 'export TEST_DEV=${part}1' >> $CONFIG_FILE");
+    assert_script_run("echo 'export SCRATCH_DEV=${part}2' >> $CONFIG_FILE");
     # Add optional mkfs options
     my $mkfs_options = get_var('XFS_MKFS_OPTIONS', '');
     $mkfs_options .= " -m reflink=1" if (get_var('XFS_TESTS_REFLINK', 0) == 1);
     $mkfs_options =~ s/^\s+|\s+$//g;    # trim string (left and right)
     assert_script_run("echo 'MKFS_OPTIONS=\"$mkfs_options\"' >> $CONFIG_FILE") unless ($mkfs_options eq '');
 }
 
+sub get_unused_nvme_device {
+    # Get all disks and partitions and return the first device, which has no parititons
+    my @disks = split /\n/, script_output('lsblk | grep disk | cut -d " " -f1');
+    my @parts = split /\n/, script_output('lsblk | grep part | cut -d " " -f1');
+    foreach my $disk (@disks) {
+        # smartmatch is still experimental, but this can be enabled in the near future.
+        #return "/dev/$disk" unless ($disk ~~ @parts);
+        return "/dev/$disk" unless (grep { m/$disk/ } @parts);
+    }
+    die "No unused nvme device available";
```

Comment:
maybe we can add `record_info` with `lsblk` output before dying? this will help debugging such cases

Reply:
Already there in line 130, which is executed before this.

Reply:
ok